core/src/main/java/org/apache/carbondata/core/indexstore/BlockletDataMapIndexStore.java

```diff
@@ -64,6 +64,13 @@
    */
   private Map<String, Object> segmentLockMap;
 
+  /**
+   * map of segmentPath to the map of index/ merge index file path to its
+   * corresponding block info. While querying the table this map will be useful while creating the
+   * block info for the index or merge index file.
+   */
+  private Map<String, Map<String, BlockMetaInfo>> segInfoCache;
```

Comment:
Global cache will be risky, how to clean it? Better have a method level cache.

Reply:
ok.. we can reuse the getAll method to reuse the method level cache.
Closing this PR